src/tool.js

```diff
@@ -0,0 +1,57 @@
+exports.genForm = (inputs) => {
+    // Create a blank string to start with
+    var formStr = '';
+    try {
+        inputs.forEach((input) => {
+            // Make sure input has property `type`
+            if (input.type) {
+                // Switch expression handles accepted input `type` props.
+                // toLowerCase makes sure we don't have to handle case sensitivity
+                switch (input.type.toLowerCase()) {
+                    // Ternary expressions in the middle of each case add context if user provides optional `id` or `for` properties
+                    case 'text':
+                        formStr += '<input type="text"' +
+                            (input.id ? ' id="' + input.id + '"' : '') +
+                            (input.name ? ' name="' + input.name + '"' : '') +
+                            ' />';
+                        break;
+                    case 'password':
+                        formStr += '<input type="password"' +
+                            (input.id ? ' id="' + input.id + '"' : '') +
+                            (input.name ? ' name="' + input.name + '"' : '') +
+                            ' />';
+                        break;
+                    case 'label':
+                        formStr += '<label' +
+                            (input.id ? ' id="' + input.id + '"' : '') +
+                            (input.for ? ' for="' + input.for+'"' : '') +
+                            '></label>';
+                        break;
+                    case 'select':
+                        formStr += '<select' +
+                            (input.id ? ' id="' + input.id + '"' : '') +
+                            '></select>';
+                        break;
+                    case 'reset':
+                        formStr += '<input type="reset"' +
+                            (input.id ? ' id="' + input.id + '"' : '') +
+                            ' />';
+                        break;
+                    case 'submit':
+                        formStr += '<input type="submit"' +
+                            (input.id ? ' id="' + input.id + '"' : '') +
+                            (input.value ? ' value="' + input.value + '"' : '') +
+                            ' />';
+                        break;
+                    default:
+                        // If input doesn't match a supported value for `type`, we don't add anything to formStr
+                        formStr += ''
```

Comment:
Would suggest adding a semicolon at end of line 48.
``` formStr += ''; ```
[ES5 semicolons](http://es5.github.io/#x7.9)
[ES6 semicolons](http://www.ecma-international.org/ecma-262/6.0/index.html#sec-automatic-semicolon-insertion)

(I know this comment is incredibly picky but your code is really solid, so far I can't find anything else)

Reply:
My only thought other than ES6 implementation would be to deliver feedback to the user here describing which submitted type could not be used, such as

> `console.log(input.type, 'type not supported.');`

Reply:
Nice catch! Fixed.